tests/test_send_graphql_query.py

```diff
@@ -24,27 +47,88 @@ def test_send_graphql_query_success(self, mock_post):
         # Assertions
         mock_post.assert_called_once_with(
             "https://platform.finitestate.io/api/v1/graphql",
-            headers={
-                'Content-Type': 'application/json',
-                'Authorization': f'Bearer {self.token}',
-                'Organization-Context': self.organization_context
-            },
-            json={'query': self.query, 'variables': self.variables}
+            headers=self.headers,
+            json={"query": self.query, "variables": self.variables},
         )
         assert result == {"data": {"result": "mock_result"}}
 
     @patch("finite_state_sdk.requests.post")
-    def test_send_graphql_query_error(self, mock_post):
+    def test_send_graphql_query_graphql_error(self, mock_post):
+        # Mock response with GraphQL errors
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {"errors": [{"message": "GraphQL error occurred"}]}
+        mock_post.return_value = mock_response
+
+        # Call the function and expect a BreakoutException
+        with pytest.raises(BreakoutException) as excinfo:
+            send_graphql_query(self.token, self.organization_context, self.query, self.variables)
+
+        assert "Error: [{'message': 'GraphQL error occurred'}]" in str(excinfo.value)
+
+    @patch("finite_state_sdk.requests.post")
+    def test_send_graphql_query_internal_server_error(self, mock_post):
         # Mock response
         mock_response = MagicMock()
         mock_response.status_code = 500
         mock_response.text = "Internal Server Error"
         mock_response.json.return_value = {"error": "Internal Server Error"}
         mock_post.return_value = mock_response
 
-        # Call the function and expect an exception
-        with pytest.raises(Exception) as excinfo:
+        # Call the function and expect a RetryError
+        with pytest.raises(tenacity.RetryError) as excinfo:
```

Comment:
if this test takes too long to run (cause of all the retries) you can add a `send_graphql_query.retry.wait = wait_fixed(0)` in the test itself to override the wait

Reply:
It only takes couple minutes.